daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -339,11 +339,13 @@ object Value extends CidContainer1[Value] {
 
   sealed abstract class ContractId extends Product with Serializable {
     def coid: String
+    def toBytes: Bytes
   }
 
   object ContractId {
     final case class V0(coid: Ref.ContractIdString) extends ContractId {
       override def toString: String = s"ContractId($coid)"
+      lazy val toBytes: Bytes = V0.prefix ++ Utf8.getBytes(coid)
```

Comment:
I'm surprised that we need another prefix. Overall, I find this API a bit confusing:
* `Ref.ContractIdString` already requires that V0 contract IDs start with a `#`. So we now have two mandatory prefixes here. Isn't `#` enough different from a `00` prefix? Do you want `toBytes` to always return a valid Base16 encoding?
* `V1`'s `toString` method includes the prefix `00`, but `V0` does not.
* The order for contract IDs decrees that `V0` comes before `V1`, but the prefixes are the opposite (`ff` vs `00`), even though the reference string itself is ordered lexicographically.

Can we make this whole thing simpler and more consistent rather than adding more on top of it?

Reply:
You are complexity right. 
I simplify by supporting only V1 in the hashing. 
Only Sandbox classic supports V0. We will see how to deal with that later, if we need the hashing in sandbox.